### PR TITLE
MM-44746_Not Open Server error when signing up with email invite

### PR DESCRIPTION
**Pull request by julmondragon**
#### Summary
When enabled email invitations, the email invite link should allow to signup.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-44746

#### Screenshots
https://user-images.githubusercontent.com/79058848/171731545-498088c3-1f8e-40a0-9350-89b6e1f98bc1.mp4

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: e2e/cypress/tests/integration/team_settings/closed_team_invite_by_email_spec.js [hunk 1]
```diff
@@ -39,6 +39,9 @@ describe('Team Settings', () => {
         cy.apiUpdateConfig({
             LdapSettings: {Enable: false},
             ServiceSettings: {EnableOnboardingFlow: true},
+            TeamSettings: {
+                EnableOpenServer: false,
+            },
```
Comment (marianunez) [inline e2e/cypress/tests/integration/team_settings/closed_team_invite_by_email_spec.js:44]: Did this test catch the issue before your fix?

File: components/signup/signup.tsx [hunk 2]
```diff
@@ -91,7 +91,7 @@ const Signup = ({onCustomizeHeader}: SignupProps) => {
     const {IsLicensed} = useSelector(getLicense);
     const loggedIn = Boolean(useSelector(getCurrentUserId));
     const useCaseOnboarding = useSelector(getUseCaseOnboarding);
-    const usedBefore = useSelector((state: GlobalState) => (!loggedIn && token ? getGlobalItem(state, token, null) : undefined));
+    const usedBefore = useSelector((state: GlobalState) => (!inviteId && !loggedIn && token ? getGlobalItem(state, token, null) : undefined));
```
Comment (marianunez) [inline components/signup/signup.tsx:94]: Is this related to this fix?

File: components/signup/signup.tsx [hunk 3]
```diff
@@ -108,7 +108,7 @@ const Signup = ({onCustomizeHeader}: SignupProps) => {
     const enableSAML = EnableSaml === 'true';
     const enableCustomBrand = EnableCustomBrand === 'true';
 
-    const noOpenServer = !inviteId && !enableOpenServer && !noAccounts;
+    const noOpenServer = !inviteId && !token && !enableOpenServer && !noAccounts;
```
Comment (marianunez) [inline components/signup/signup.tsx:111]: How could we test that the error is still showing properly for no open server?

File: e2e/cypress/tests/integration/team_settings/closed_team_invite_by_email_spec.js (hunk 1 above)
Comment (julmondragon) [reply e2e/cypress/tests/integration/team_settings/closed_team_invite_by_email_spec.js:44]: Correct.

**Review by marianunez — APPROVED** [review]
Thanks for the quick fix on this @julmondragon! 💯

File: e2e/cypress/tests/integration/team_settings/closed_team_invite_by_email_spec.js (hunk 1 above)
Comment (marianunez) [reply e2e/cypress/tests/integration/team_settings/closed_team_invite_by_email_spec.js:44]: I see, that makes sense. Thanks!

**Review by stevemudie — APPROVED** [review]
Worked great! This is all set @julmondragon

**Comment by amyblais** [comment]
@julmondragon This needs a manual cherry-pick to ``release-7.0``.

**Comment by julmondragon** [comment]
> @julmondragon This needs a manual cherry-pick to `release-7.0`.

@amyblais https://github.com/mattermost/mattermost-webapp/pull/10472
